Add blog with Airflow Survey results (#219)
apache/airflow-site: landing-pages/site/content/en/blog/airflow-survey/index.md

**Do you combine multiple DAGs?**

|                                 |No.|  %   |
|---------------------------------|---|------|
|No, I don't combine multiple DAGs|127|41.23%|
|Yes, through SubDAG              | 73|23.70%|
|Yes, by triggering another DAG   | 72|23.38%|
|Other                            | 36|11.69%|

In the other category, 9 people explicitly mentioned using `ExternalTaskSensor`,
and I think it could be treated as running subDAGs by triggering other DAGs.

**Do you use Airflow Plugins? If yes, what do you use it for?**

|                                      |No.|  %   |
|--------------------------------------|---|------|
|Adding new operators/sensors and hooks|187|60.71%|
|I don't use Airflow plugins           |109|35.39%|
|Adding AppBuilder views & menu items  | 31|10.06%|
|Adding new executor                   | 18|5.84% |
|Adding OperatorExtraLinks             |  7|2.27% |

The high percentage - 60%  for “Adding new operators/sensors and hooks” is quite a
surprising result for some of us - especially that you do not actually need to use the
plugin mechanism to add any of those. Those are standard python objects, and you can

> simply drop your hooks/operators/sensors code to PYTHONPATH and they will work.

Reviewer: ```suggestion
simply drop your hooks/operators/sensors code to `PYTHONPATH` environment variable and they will work.
```

It seems that this may be a result of a lack of best practices guide.

Plugins are more useful for adding views and menu items - yet only 10%.
OperatorExtraLinks are even more useful (though relatively new) feature, so it’s not
entirely surprising they are hardly used.

It was also kind of surprising that someone at all uses plugins to use their own
executors. We considered removing that option recently - but now we have to rethink
our approach.

**What metrics do you use to monitor Airflow?**

There were a lot of different responses. Some use Prometheus and other services,
others do not use any monitoring. One of the interesting responses linked to this

> solution: https://github.com/mastak/airflow_operators_metrics

Reviewer: Can you remove pure link and replace it with anchor e.g. `[airflow_operators_metrics](https://github.com/mastak/airflow_operators_metrics)`?  Each link should describe the content.
https://uxmovement.com/content/why-your-links-should-never-say-click-here/ 
https://www.w3.org/QA/Tips/noClickHere

## External services

**What external services do you use in your Airflow DAGs?**

|                                                 |No.|  %   |
|-------------------------------------------------|---|------|
|Amazon Web Services                              |160|51.95%|
|Internal company systems                         |150|48.7% |
|Hadoop / Spark / Flink / Other Apache software   |119|38.64%|
|Google Cloud Platform / Google APIs              |112|36.36%|
|Microsoft Azure                                  | 28|9.09% |
|I do not use external services in my Airflow DAGs| 18|5.84% |

> It’s not surprising that AWS is leading the way as they are considered the most mature

Reviewer: ```suggestion
It’s not surprising that Amazon Web Service is leading the way as they are considered the most mature
```

services are equally popular.

## What can be improved

**In your opinion, what could be improved in Airflow?**

|                                                  |No.|  %   |
|--------------------------------------------------|---|------|
|Scheduler performance                             |189|61.36%|
|Web UI                                            |180|58.44%|
|Logging, monitoring and alerting                  |145|47.08%|
|Examples, how-to, onboarding documentation        |143|46.43%|
|Technical documentation                           |137|44.48%|
|Reliability                                       |112|36.36%|
|REST API                                          | 96|31.17%|
|Authentication and authorization                  | 89|28.9% |
|External integration e.g. AWS, GCP, Apache product| 49|15.91%|
|CLI                                               | 41|13.31%|
|I don’t know                                      |  5|1.62% |

The results are rather quite self-explaining. Improved performance of Airflow, better
UI, and more telemetry are desirable. But this should go in pair with improved
documentation and resources about using the Airflow, especially when we

> take into account the problem of onboarding new users (see earlier).

Reviewer: ```suggestion
take into account the problem of onboarding new users.
```
